delocate/delocating.py

```diff
@@ -526,42 +558,44 @@ def delocate_wheel(
         all_copied = {}  # type: Dict[Text, Dict[Text, Text]]
         wheel_dir = realpath(pjoin(tmpdir, 'wheel'))
         zip2dir(in_wheel, wheel_dir)
-        for package_path in find_package_dirs(wheel_dir):
-            lib_path = pjoin(package_path, lib_sdir)
-            lib_path_exists = exists(lib_path)
-            copied_libs = delocate_path(
-                package_path,
-                lib_path,
-                lib_filt_func,
-                copy_filt_func,
-                executable_path=executable_path,
-                ignore_missing=ignore_missing,
-            )
-            if copied_libs and lib_path_exists:
+        # Assume the package name from the wheel filename.
+        package_name = basename(in_wheel).split("-")[0]
+        lib_sdir = _decide_dylib_bundle_directory(
+            wheel_dir, package_name, lib_sdir
+        )
+        lib_path = pjoin(wheel_dir, lib_sdir)
+        lib_path_exists = exists(lib_path)
+        copied_libs = delocate_path(
+            wheel_dir,
+            lib_path,
+            lib_filt_func,
+            copy_filt_func,
+            executable_path=executable_path,
+            ignore_missing=ignore_missing,
+        )
+        if copied_libs and lib_path_exists:
+            raise DelocationError(
+                '{0} already exists in wheel but need to copy '
+                '{1}'.format(lib_path, '; '.join(copied_libs)))
+        if len(os.listdir(lib_path)) == 0:
+            shutil.rmtree(lib_path)
+        # Check architectures
+        if require_archs is not None:
+            stop_fast = not check_verbose
+            bads = check_archs(copied_libs, require_archs, stop_fast)
+            if len(bads) != 0:
+                if check_verbose:
+                    print(bads_report(bads, pjoin(tmpdir, 'wheel')))
```

Comment:
Should this be a logger call now?

Reply:
I don't know.  It is kind of weird to optionally report error information like this when an exception is going to be raised anyway.  I might've included this in the exception message.

Reply:
OK - how about appending this to the error message instead?

Reply:
The current behavior is according to the documentation.  There isn't much that can be done here.

Reply:
It's really not important, feel free to leave as is.  If, by documentation, you just meant the docstring, then I think adding it to the error message ends up with just as much "printing" as using `print`, so I don't worry about that myself.

Reply:
Logging sends to `stderr` instead of `stdout` and changes the format.  Using print like this is technically a contract with other CLI tools which is why I'm having a hard time with it.

Since you didn't intend it to be that way then I'll be okay with breaking it.

Reply:
I kind of want to ignore `check_verbose` and always print the full error if an issue is found.

Reply:
Yes, that's fine too.